meta-cli/src/deploy/actors/task/list.rs

```diff
@@ -0,0 +1,181 @@
+pub struct ListSuccess {
+    pub typegraph: String,
+}
+
+impl OutputData for ListSuccess {
+    fn get_typegraph_name(&self) -> String {
+        self.typegraph.clone()
+    }
+    fn is_success(&self) -> bool {
+        true
+    }
+}
+
+#[derive(Deserialize, Debug)]
+pub struct ListError {
+    pub typegraph: String,
+    pub errors: Vec<String>,
+}
+
+#[derive(Debug, Default)]
+pub struct ListOptions {
+    filter: TaskFilter,
+}
+
+impl OutputData for ListError {
+    fn get_typegraph_name(&self) -> String {
+        self.typegraph.clone()
+    }
+    fn is_success(&self) -> bool {
+        false
+    }
+}
+
+impl TaskAction for ListAction {
+    type SuccessData = ListSuccess;
+    type FailureData = ListError;
+    type Options = ListOptions;
+    type Generator = ListActionGenerator;
+    type RpcCall = serde_json::Value;
+
+    async fn get_command(&self) -> Result<Command> {
+        build_task_command(
+            self.task_ref.path.clone(),
+            self.shared_config.clone(),
+            self.task_options.filter.clone(),
+        )
+        .await
+    }
+
+    fn get_options(&self) -> &Self::Options {
+        &self.task_options
+    }
+
+    fn get_start_message(&self) -> String {
+        format!(
+            "starting list process for {:?}",
+            self.task_ref.path.display().yellow()
+        )
+    }
+
+    fn get_error_message(&self, err: &str) -> String {
+        format!(
+            "{icon} failed to liste typegraph(s) from {path:?}: {err}",
+            icon = "✗".red(),
+            path = self.task_ref.path.display().yellow(),
+            err = err,
+        )
+    }
+
+    async fn finalize(
+        &self,
+        res: &ActionResult<Self>,
+        ctx: ActionFinalizeContext<Self>,
+    ) -> Result<Option<Box<dyn FollowupOption<ListAction>>>> {
+        match res {
+            Ok(data) => {
+                ctx.console.info(format!(
+                    "{icon} successfully listed typegraph {name} from {path}",
+                    icon = "✓".green(),
+                    name = data.get_typegraph_name().cyan(),
+                    path = self.task_ref.path.display().yellow(),
+                ));
+            }
+            Err(output) => {
+                ctx.console.error(format!(
+                    "{icon} failed to liste typegraph {name} from {path}",
+                    icon = "✗".red(),
+                    name = output.get_typegraph_name().cyan(),
+                    path = self.task_ref.path.display().yellow(),
+                ));
+                for err in output.errors.iter() {
+                    ctx.console.error(format!("- {err}"));
+                }
+            }
+        }
+
+        Ok(None)
+    }
+
+    fn get_task_ref(&self) -> &crate::deploy::actors::task_manager::TaskRef {
+        &self.task_ref
+    }
+
+    async fn get_rpc_response(&self, _call: &serde_json::Value) -> Result<serde_json::Value> {
+        Err(ferr!("rpc request not supported on serialize task"))
```

Comment:
"list"

Reply:
I think is diff between my last commit and my branch now with rebase onto main